### PR TITLE
feat(@desktop/wallet): Implement Network selection in SwapModal

### LINKED ISSUE #14750: Network selection
https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3404-111628&t=mcZdgp5PUHsBgxv7-4

Adding network selection for Swap and also linking it with launch flows

**Acceptance Criteria**: 

https://www.notion.so/Select-network-a8d2aeeeec884558b52cd00eff6a8706?pvs=4
most of the things mentioned here will be acceptance criteria for this task https://github.com/status-im/status-desktop/issues/14780 not the current one as we are going to use an existing component here that doesn't match the design 100% 

but from integration perspective the acceptance criteria for this task should be:
1. When clicking on the networks in the list, it should become the selected one
2. This selected network should effect what we see in the accounts modal header drop down, so that the balances for select asset on that network in now shown.

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #14750

### What does the PR do

Uses the existing Network filter in its current state in the Swap Modal.
Added Unit tests for the interactions with the network selector in context to the SwapModal

Unit testing the new Network filter should be done under https://github.com/status-im/status-desktop/issues/14780

### Affected areas

SwapModal

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it
Design does match for network filter as of now, will be done under https://github.com/status-im/status-desktop/issues/14780

https://github.com/status-im/status-desktop/assets/60327365/c6f82acf-e5b1-4d55-a5a3-07965da33c83

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml [hunk 1]
```diff
@@ -19,14 +21,12 @@ StatusDialog {
     required property SwapModalAdaptor swapAdaptor
 
     objectName: "swapModal"
-    title: qsTr("Swap")
 
+    implicitWidth: 556
     bottomPadding: 16
     padding: 0
 
     background: StatusDialogBackground {
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:29]: You can just use `backgroundColor` on the StatusDialog to customize the color; no need to overrride the whole `background`

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml [hunk 2]
```diff
@@ -46,27 +46,60 @@ StatusDialog {
     }
 
     // This is a temporary placeholder while each of the components are  being added.
-    contentItem: Column {
+    ColumnLayout {
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:49]: why not `contentItem` + `padding` instead of margins here?

**Review by caybro — APPROVED** [review]
LGTM

Just some minor stuff inline

**Review by alexjba — APPROVED** [review]
LGTM!

**Pull request merged**